### PR TITLE
feat: policy-config flag

### PR DESCRIPTION
**Pull request by dimabru**

### CODE REVIEW

File: bl/evaluation/evaluator_test.go [hunk 1]
```diff
@@ -276,7 +276,7 @@ func newFilesConfigurations(path string) []*extractor.FileConfigurations {
 }
 
 func mockGetPreRunData() *cliClient.EvaluationPrerunDataResponse {
-	const policiesJsonPath = "../../internal/fixtures/policyAsCode/policies.json"
+	const policiesJsonPath = "../../internal/fixtures/policyAsCode/prerun.json"
```
Comment (TzlilSwimmer123) [inline bl/evaluation/evaluator_test.go:279]: 💯

File: bl/policy/policy_factory.go [hunk 2]
```diff
@@ -60,6 +68,40 @@ func CreatePolicy(policies *cliClient.EvaluationPrerunPolicies, policyName strin
...
+	fileReader := fileReader.CreateFileReader(nil)
+	policiesStr, err := fileReader.ReadFileContent(path)
+	if err != nil {
+		return nil, err
+	}
+	validationResults, err := validatePoliciesYaml(policiesStr)
+	if err != nil {
+		return nil, err
+	}
+	if !validationResults.Valid() {
```
Comment (alexfedin) [inline bl/policy/policy_factory.go]: This should be part of the validatepoliciesYaml function because then you have 2 places that are responsible for validation, the `validatePoliciesYaml` function and the if scope in this function

File: bl/policy/policy_factory.go [hunk 3]
```diff
@@ -60,6 +68,40 @@ func CreatePolicy(policies *cliClient.EvaluationPrerunPolicies, policyName strin
 	return Policy{policyName, rules}, nil
 }
 
+func GetPoliciesFileFromPath(path string) (*cliClient.EvaluationPrerunPolicies, error) {
+	fileReader := fileReader.CreateFileReader(nil)
+	policiesStr, err := fileReader.ReadFileContent(path)
```
Comment (alexfedin) [inline bl/policy/policy_factory.go]: To make this function cleaner, so it will be responsible only for one thing, you should extract reading the file outside and then validate the contents of the file outside. After that you can pass the contents to `GetPoliciesFileFromPath` function so it will only be responsible for extracting the policies, as the name of the functions suggests

File: pkg/jsonSchemaValidator/validator.go [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+	json, _ := yaml.YAMLToJSON([]byte(yamlContent))
+
+	schemaLoader := gojsonschema.NewStringLoader(string(jsonSchema))
+	documentLoader := gojsonschema.NewStringLoader(string(json))
+
+	return gojsonschema.Validate(schemaLoader, documentLoader)
+}
+
+// func (jsv *JSONSchemaValidator) Validate(jsonSchema map[string]interface{}, yamlContent string) (*Result, error) {
```
Comment (alexfedin) [inline pkg/jsonSchemaValidator/validator.go]: remove redundant comment

File: pkg/jsonSchemaValidator/validator.go [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+	return &JSONSchemaValidator{}
+}
+
+type Result = gojsonschema.Result
+
+func (jsv *JSONSchemaValidator) ValidateYamlSchema(schemaContent string, yamlContent string) (*Result, error) {
+
+	jsonSchema, _ := yaml.YAMLToJSON([]byte(schemaContent))
+
+	json, _ := yaml.YAMLToJSON([]byte(yamlContent))
```
Comment (alexfedin) [inline pkg/jsonSchemaValidator/validator.go]: Use Validate function here to avoid code duplication

**Review by alexfedin — CHANGES_REQUESTED** [review]

File: pkg/cliClient/cliClient_test.go [hunk 6]
```diff
@@ -342,7 +342,7 @@ func test_getVersionMessage_success() *GetVersionMessageTestCase {
 }
```
Comment (royhadad) [inline pkg/cliClient/cliClient_test.go:342]: why did you rename policies.json to prerun.json?

Comment (dimabru) [reply pkg/cliClient/cliClient_test.go:342]: because the file is actually a mock of the response of the `prerun` call to the backend

File: bl/policy/policy_factory.go (hunk 2 above)
Comment (dimabru) [reply bl/policy/policy_factory.go]: done

File: bl/policy/policy_factory.go (hunk 3 above)
Comment (dimabru) [reply bl/policy/policy_factory.go]: @MentalBrake 
what if I change the name of the function to `populatePolicies(path string)`?

Comment (alexfedin) [reply bl/policy/policy_factory.go]: Then you still get a function that is responsible for reading a file, validating a yaml against json schema and parsing it into policies, this creates a coupling between multiple functionalities

Comment (dimabru) [reply bl/policy/policy_factory.go]: I prefer to keep the logic of getting the policy post-validation inside the policy_factory package. I don't mind splitting the `GetPoliciesFileFromPath` into several functions with one 'above' that will call the file creation, conversion and validation.
What should I name the function inside `policy_factory`?
I want to solve this without having to add the validation and file retrieving logic to the caller function (which is the `test` command)

Comment (alexfedin) [reply bl/policy/policy_factory.go]: we don't have to make these changes, it can be refactored later, anyway this function does not belong to a policy factory, the factory should only be responsible for creating objects, move this to policy package

**Review by alexfedin — DISMISSED** [review]

**Review by alexfedin — APPROVED** [review]
